packages/react-native/src/surveys/getActiveMatchingSurveys.ts

```diff
@@ -97,27 +103,27 @@ export function getActiveMatchingSurveys(
       return true
     }
 
-    const linkedFlagCheck = survey.linked_flag_key ? !!flags[survey.linked_flag_key] === true : true
+    const linkedFlagCheck = isSurveyFlagEnabled(survey.linked_flag_key, flags)
 
     const linkedFlagVariant = survey.conditions?.linkedFlagVariant
     let linkedFlagVariantCheck = true
     if (linkedFlagVariant) {
       linkedFlagVariantCheck = survey.linked_flag_key
-        ? flags[survey.linked_flag_key] === linkedFlagVariant || linkedFlagVariant === 'any'
+        ? flags[survey.linked_flag_key] === linkedFlagVariant || linkedFlagVariant === ANY_FLAG_VARIANT
         : true
     }
 
-    const targetingFlagCheck = survey.targeting_flag_key ? !!flags[survey.targeting_flag_key] === true : true
+    const targetingFlagCheck = isSurveyFlagEnabled(survey.targeting_flag_key, flags)
 
     const eventBasedTargetingFlagCheck = hasEvents(survey) ? activatedSurveys.has(survey.id) : true
 
     const internalTargetingFlagCheck =
       survey.internal_targeting_flag_key && !canActivateRepeatedly(survey)
-        ? !!flags[survey.internal_targeting_flag_key] === true
+        ? isSurveyFlagEnabled(survey.internal_targeting_flag_key, flags)
         : true
```

Comment:
`const internalTargetingFlagCheck = isSurveyFlagEnabled(survey.internal_targeting_flag_key, flags) || canActivateRepeatedly(survey)`

can be simplified I think?